### PR TITLE
docs: link and mention optional features

### LINKED ISSUE #618: "Failed to create Kubernetes secret error" when trying to sync AWS Secrets Manager to Kubernetes Secrets
**What steps did you take and what happened:**
When doing the setup using Helm and syncing the Secrets from SPC to Kubernetes Secrets I am receiving the error message 
"secretproviderclasspodstatus_controller.go:325] "failed to create Kubernetes secret" err="secrets is forbidden: User \"system:serviceaccount:kube-system:secrets-store-csi-driver\" cannot create resource \"secrets\" in API group \"\" in the namespace \"default\"" spc="default/aws-secrets" pod="default/pod" secret="default/secret" spcps="default/pod-deployment-default-aws-secrets""

**What did you expect to happen:**
CSI Secrets Store Driver sync Secrets object to Kubernetes Secrets.

**Anything else you would like to add:**
To fix this I just added the permission below to the Cluster Role "[secretproviderclasses-role](https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/charts/secrets-store-csi-driver/templates/role.yaml)"
```
- apiGroups:
  - ""
  resources:
  - secrets 
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
```
**Which provider are you using:**
AWS

**Environment:**

- Secrets Store CSI Driver version: v0.0.23
- Kubernetes version: 1.19

### PR DESCRIPTION
**Pull request by tam7t**
**What this PR does / why we need it**:

More prominently display optional parameters when installing the CSI driver that will be commonly used.

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #618

**Special notes for your reviewer**:

I also alphabetized the list of providers

### CODE REVIEW

File: docs/book/src/getting-started/installation.md [hunk 1]
```diff
@@ -25,7 +25,14 @@ helm install csi-secrets-store secrets-store-csi-driver/secrets-store-csi-driver
...
 namespace.
 
-#### Values
+#### Optional Values
+
+Notably the following feature must be explicitly enabled:
+
+| Feature | Helm Parameter |
+|-----|-----|
+| [Sync as Kubernetes secret](./topics/sync-as-kubernetes-secret.md)) | `syncSecret.enabled=true`|
```
Comment (aramase) [inline docs/book/src/getting-started/installation.md:34]:
```suggestion
| [Sync as Kubernetes secret](../topics/sync-as-kubernetes-secret.md) | `syncSecret.enabled=true`|
```

File: docs/book/src/getting-started/installation.md [hunk 2]
```diff
@@ -25,7 +25,14 @@ helm install csi-secrets-store secrets-store-csi-driver/secrets-store-csi-driver
...
+| [Secret Auto rotation](./topics/secret-auto-rotation.md) | `enableSecretRotation=true`|
```
Comment (aramase) [inline docs/book/src/getting-started/installation.md:35]:
```suggestion
| [Secret Auto rotation](../topics/secret-auto-rotation.md) | `enableSecretRotation=true`|
```

File: docs/book/src/getting-started/installation.md [hunk 3]
```diff
@@ -72,11 +79,11 @@ secretproviderclasses.secrets-store.csi.x-k8s.io
 secretproviderclasspodstatuses.secrets-store.csi.x-k8s.io
 ```
 
-## Use the Secrets Store CSI Driver with a Provider
+## Install External Secret Providers
 
 Now that the Secrets Store CSI Driver has been deployed, select a provider from the supported provider list, then follow the installation steps for the provider:
 
+- [AWS Provider](https://github.com/aws/secrets-store-csi-driver-provider-aws)
```
Comment (aramase) [inline docs/book/src/getting-started/installation.md:86]: Do you want to update the order and links here too?

https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/docs/book/src/introduction.md
https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/docs/book/src/concepts.md

Comment (tam7t) [reply docs/book/src/getting-started/installation.md:86]: sgtm

**Review by aramase — COMMENTED** [review]
/lgtm

**Pull request merged**